statemachine/statemachine.py

```diff
@@ -277,6 +277,23 @@ def __repr__(self):
             self.current_state.identifier,
         )
 
+    def _visit_neighbour_states(self, transition, visited_states):
+        for neighbour_state in transition.destinations:
+            if neighbour_state not in visited_states:
+                self._visitable_states(neighbour_state, visited_states)
+
+    def _visitable_states(self, start_state, visited_states):
+        visited_states.append(start_state)
+
+        for transition in start_state.transitions:
+            self._visit_neighbour_states(transition, visited_states)
+        return visited_states
+
+    def _disconnected_states(self, starting_state):
+        visitable_states = self._visitable_states(starting_state, visited_states=[])
+
+        return set(self.states) - set(visitable_states)
```

Comment:
@fgmacedo I've decided to return just the difference between the actual number of states and the states visitable from the initial state. I think that's a good design because an empty set is Falsy, which suits good for verification using `if`.